sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/MethodsShouldNotHaveTooManyLines.cs

```diff
@@ -34,86 +33,60 @@ namespace SonarAnalyzer.Rules.CSharp
-                c =>
-                {
-                    if (Max < 2)
-                    {
-                        throw new ArgumentOutOfRangeException(
-                            $"Invalid rule parameter: maximum number of lines = {Max}. Must be at least 2.");
-                    }
-
-                    var baseMethodSyntax = (BaseMethodDeclarationSyntax)c.Node;
-
-                    var identifierLocation = baseMethodSyntax?.FindIdentifierLocation();
-                    if (identifierLocation == null)
-                    {
-                        return;
-                    }
+        protected override GeneratedCodeRecognizer GeneratedCodeRecognizer =>
+            Helpers.CSharp.CSharpGeneratedCodeRecognizer.Instance;
 
-                    var linesCount = GetBodyTokens(baseMethodSyntax)
-                        .SelectMany(token => token.GetLineNumbers())
-                        .Distinct()
-                        .LongCount();
-
-                    if (linesCount > Max)
-                    {
-                        c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, identifierLocation,
-                            GetDescription(baseMethodSyntax), linesCount, Max));
-                    }
-                },
+        protected override SyntaxKind[] SyntaxKinds { get; } =
+            new[]
+            {
                 SyntaxKind.MethodDeclaration,
                 SyntaxKind.ConstructorDeclaration,
-                SyntaxKind.DestructorDeclaration);
-        }
+                SyntaxKind.DestructorDeclaration
+            };
 
-        private static IEnumerable<SyntaxToken> GetBodyTokens(BaseMethodDeclarationSyntax baseMethodSyntax) =>
-            baseMethodSyntax?.ExpressionBody()?.Expression?.DescendantTokens()
-            ?? baseMethodSyntax?.Body?.Statements.SelectMany(s => s.DescendantTokens())
-            ?? Enumerable.Empty<SyntaxToken>();
+        protected override string MethodKeyword { get; } = "methods";
 
-        private static string GetDescription(BaseMethodDeclarationSyntax baseMethodDeclaration)
+        protected override IEnumerable<SyntaxToken> GetMethodTokens(BaseMethodDeclarationSyntax baseMethodDeclaration) =>
+            baseMethodDeclaration?.ExpressionBody()?.Expression?.DescendantTokens()
+                ?? baseMethodDeclaration?.Body?.Statements.SelectMany(s => s.DescendantTokens())
+                ?? Enumerable.Empty<SyntaxToken>();
+
+        protected override SyntaxToken? GetMethodIdentifierToken(BaseMethodDeclarationSyntax baseMethodDeclaration) =>
+            baseMethodDeclaration?.GetIdentifierOrDefault();
+
+        protected override string GetMethodKindAndName(SyntaxToken identifierToken)
         {
-            var identifierName = baseMethodDeclaration.GetIdentifierOrDefault()?.ValueText;
+            var identifierName = identifierToken.ValueText;
             if (identifierName == null)
             {
                 return "method";
```

Comment:
is this covered with a test?

Reply:
you should also add a test for C# with NullOrEmpty identifier name (fix the leak!)

Reply:
@andrei-epure-sonarsource I don't see why this is showing `== null` because the code is `NullOrEmpty` check.